crates/data/src/lib.rs

```diff
@@ -117,3 +117,75 @@ pub struct Summary<'a> {
     /// The mean deviation (note: not standard deviation) of the `count` field.
     pub mean_deviation: f64,
 }
+
+/// The effect size (and confidence interval) of between two different engines
+/// (i.e. two different commits of Wasmtime).
+///
+/// This allows us to justify statements like "we are 99% confident that the new
+/// register allocator is 13.6% faster (± 1.7%) than the old register
+/// allocator."
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct EffectSize<'a> {
+    /// The CPU architecture on which this measurement was taken, for example
+    /// "aarch64" or "x86_64".
+    pub arch: Cow<'a, str>,
+
+    /// The file path of the Wasm benchmark program.
+    pub wasm: Cow<'a, str>,
+
+    /// The phase in a Wasm program's lifecycle that was measured: compilation,
+    /// instantiation, or execution.
+    pub phase: Phase,
+
+    /// The event that was measured: micro seconds of wall time, CPU cycles
+    /// executed, instructions retired, cache misses, etc.
+    pub event: Cow<'a, str>,
+
+    /// The first engine variant.
+    pub a: EffectSizeVariant<'a>,
+
+    /// The second engine variant.
+    pub b: EffectSizeVariant<'a>,
```

Comment:
Not sure if the multi-level structuring here will show up in the CSV output. Do you know? I feel like sometime ago we switched to single-level output structs for this reason. (We could add a test in `crates/data/tests` to check this?).

Reply:
Good call, I forgot that nested structs aren't automatically flattened with the `csv` crate. Pushed a fix.